2.2.0/docker-entrypoint.sh

```diff
@@ -25,13 +25,26 @@ if [ "$1" = 'couchdb' ]; then
 fi
 
 if [ "$1" = '/opt/couchdb/bin/couchdb' ]; then
-	# we need to set the permissions here because docker mounts volumes as root
-	chown -fR couchdb:couchdb /opt/couchdb || true
+	# Check that we own everything in /opt/couchdb and fix if necessary
+	find /opt/couchdb \! \( -user couchdb -group couchdb \) -exec chown couchdb:couchdb '{}' +
```

Comment:
Needs a `-f`

Reply:
So ... I want to test that one a bit. I've already scoped down the chown so it only executes on files that have incorrect ownership. I'm struggling to think of a case where we have files in /opt/couchdb that are not owned by couchdb and yet we're OK to proceed without that ownership change. I guess the best example would be a config file with 0644 permissions owned by root coming in from a volume mount?

Reply:
Right, the big issues are config and data files owned by the wrong person. We could conceivably skip changing their ownership as long as we have read/write access....

Another option is to have Couch itself complain when it sees config files it can't read/write, skip all the checks here, and let Couch crash out. We don't do that yet.

It'd be harder to do that for data files since generally we don't start opening db/view files until asked to do so by the user.